test/aqua/test_phase_estimator.py

```diff
@@ -0,0 +1,214 @@
+from qiskit.aqua.algorithms.phase_estimators.hamiltonian_pe import TempPauliEvolve
+from qiskit.aqua.operators.evolutions import PauliTrotterEvolution, MatrixEvolution
+import qiskit
+from qiskit.aqua.operators import (H, X, Y, Z, I)
+
+
+class TestHamiltonianPE(QiskitAquaTestCase):
+    """Tests for obtaining eigenvalues from phase estimation"""
+
+    def setUp(self):
+        super().setUp()
+        self.hamiltonian_1 = (0.5 * X) + Y + Z
+
+    def hamiltonian_pe(self, hamiltonian, state_preparation=None, num_evaluation_qubits=6,
+                       backend=qiskit.Aer.get_backend('qasm_simulator'),
+                       evolution=TempPauliEvolve()):
+        """Run HamiltonianPE and return result with all  phases."""
+        quantum_instance = qiskit.aqua.QuantumInstance(backend=backend, shots=100000)
+        phase_est = HamiltonianPE(
+            num_evaluation_qubits=num_evaluation_qubits,
+            hamiltonian=hamiltonian, quantum_instance=quantum_instance,
+            state_preparation=state_preparation, evolution=evolution)
+        result = phase_est.run()
+        return result
+
+    # pylint: disable=invalid-name
+    def test_pauli_sum_1(self):
+        """Two eigenvalues from Pauli sum with X, Z"""
+        a1 = 0.5
+        a2 = 1.0
+        hamiltonian = (a1 * X) + (a2 * Z)
+        state_preparation = H.to_circuit()
+        result = self.hamiltonian_pe(hamiltonian, state_preparation)
+        phase_dict = result.filter_phases(0.162, as_float=True)
+        phases = list(phase_dict.keys())
+        self.assertAlmostEqual(phases[0], 1.125, delta=0.001)
+        self.assertAlmostEqual(phases[1], -1.125, delta=0.001)
+
+    def test_pauli_sum_2(self):
+        """Two eigenvalues from Pauli sum with X, Y, Z"""
+        hamiltonian = self.hamiltonian_1
+        state_preparation = None
+        result = self.hamiltonian_pe(hamiltonian, state_preparation)
+        phase_dict = result.filter_phases(0.1, as_float=True)
+        phases = list(phase_dict.keys())
+        self.assertAlmostEqual(phases[0], 1.484, delta=0.001)
+        self.assertAlmostEqual(phases[1], -1.484, delta=0.001)
+
+    def test_matrix_evolution(self):
+        """1Q Hamiltonian with MatrixEvolution"""
+        hamiltonian = (0.5 * X) + (0.6 * Y) + (0.7 * I)
+        state_preparation = None
+        result = self.hamiltonian_pe(hamiltonian, state_preparation, evolution=MatrixEvolution())
+        phase_dict = result.filter_phases(0.2, as_float=True)
+        phases = list(phase_dict.keys())
+        self.assertAlmostEqual(phases[0], 0.0562, delta=0.001)
+        self.assertAlmostEqual(phases[1], -1.462, delta=0.001)
+
+    def _setup_from_bound(self, evolution):
+        hamiltonian = self.hamiltonian_1
+        state_preparation = None
+        bound = 1.2 * sum([abs(hamiltonian.coeff * pauli.coeff) for pauli in hamiltonian])
+        backend = qiskit.Aer.get_backend('qasm_simulator')
```

Comment:
Can we use BasicAer in the tests unless its needed to use Aer for some reason, e.g its much faster. If we need to use Aer then it we should add the skip test logic (see other tests that use Aer) in case its not installed.